### PR TITLE
Check if custom setting is used

### PR DESCRIPTION
**Pull request by KristjanZigart**
## Description

This PR includes changes to code documentation and two functions for a specific use-case
if a user needs to know if custom or default settings are in use.

## Areas of interest for the reviewer

Folders:
- user_settings
- tests

## Checklist

- [x] My code follows the [style guidelines] as defined by IRNAS.
- [x] I have performed a self-review of my code.
- [x] My changes generate no new warnings.
- [x] I added/updated source code documentation for all newly added or changed functions.
- [x] I updated the CHANGELOG.
- [x] ~~I updated all customer-facing technical documentation.~~ Only internal changes.

## After-review steps

* Reviewer can merge and delete the branch.

### CODE REVIEW

**Comment by KristjanZigart** [comment]
* **#54** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-usersettings-lib/54?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @KristjanZigart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by TjazVracko — CHANGES_REQUESTED** [review]
This is a good start.
Remember what we discussed in person - you have added the missing functions, but the API now does not make much sense (not your fault :)).

What you still have to do:

1. Add a test for `user_settings_is_set_with_id/key` that tests the desired behaviour. The desired behaviour is: `This function returns true if a default or custom value for this setting exists.` If the tests fail (which they should), fix the implementation.
2. Rename `user_settings_has_default_with_id/key` to `user_settings_is_default_set_with_id/key` (to match the naming of the other two). 
3.  Add a test for `user_settings_is_default_set_with_id/key` that tests the desired behaviour. The desired behaviour is: `This function returns true if a default is set. It does not care about the custom value` 

This way we will have all 3 possible "checks":
1. Does the setting have a value I can use (is_set)
2. Does the setting have a default value (is_default_set)
3. Does the setting have a non-default value (is_custom_set)

File: tests/user_settings/src/main.c [hunk 1]
```diff
@@ -446,6 +446,95 @@ ZTEST(user_settings_suite, test_settings_user_settings_any_changed)
 	c = user_settings_any_changed();
 	zassert_false(c, "No setting should be marked changed");
 }
+
+ZTEST(user_settings_suite, test_settings_is_custom_set)
```
[the remark this answers is not in the document]
Comment (TjazVracko) [reply tests/user_settings/src/main.c:450]: Taking another look at this...
Just delete all other data types and only have a test using Strings - the rest is redundant.

You also do not test all possibilities:
1. No default, no value
2. set default, no value
3. no default, set value
4. set default, set value

Modify the test so it sets up all 4 possibilities and then check the return of `user_settings_is_custom_set_with_id`.

**Review by TjazVracko — CHANGES_REQUESTED** [review]

File: tests/user_settings/src/main.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (TjazVracko) [reply tests/user_settings/src/main.c:450]: Taking another look at this...
Just delete all other data types and only have a test using Strings - the rest is redundant.

You also do not test all possibilities:

No default, no value
set default, no value
no default, set value
set default, set value
Modify the test so it sets up all 4 possibilities and then check the return of `user_settings_is_custom_set_with_id`.

**Comment by TjazVracko** [comment]
Work is stale and not used, closing.

**Pull request closed without merging**